memory/transparent_hugepages_defrag.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Test Inspired basically from
+# https://github.com/autotest/tp-qemu/blob/master/generic/tests/trans_hugepage_defrag.py
+#
+# Copyright: 2017 IBM
+# Author: Santhosh G <santhog4@linux.vnet.ibm.com>
+
+import os
+import time
+import mmap
+import avocado
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import memory
+from avocado.utils import disk
+from avocado.core import data_dir
+from avocado.utils.partition import Partition
+
+
+class Thp_Defrag(Test):
+
+    '''
+    Defrag test enables THP and fragments the system memory using dd load
+    and turns on THP defrag and checks whether defrag occured.
+    '''
+
+    def setUp(self):
+
+        '''
+        Sets required params for dd workload and mounts the tmpfs
+        '''
+
+        # Get required mem info
+        self.mem_path = os.path.join(data_dir.get_tmp_dir(), 'thp_space')
+        self.block_size = int(mmap.PAGESIZE) / 1024
+        self.dd_timeout = self.params.get('dd_timeout', default=900)
```

Comment:
`self.dd_timeout`  no need to fetch through `params.get` as test not have any input from yaml